cmake/FindPicoSAT.cmake

```diff
@@ -4,7 +4,11 @@
 # PicoSAT_LIBRARIES - Libraries needed to use PicoSAT
 
 find_path(PicoSAT_INCLUDE_DIR NAMES picosat.h)
-find_library(PicoSAT_LIBRARIES NAMES picosat)
+if(NOT SHARED)
+  find_library(PicoSAT_LIBRARIES NAMES libpicosat.a picosat)
```

Comment:
This should not be necessary since we already prefer .a libraries if we don't build shared (https://github.com/Boolector/boolector/blob/master/CMakeLists.txt#L169).

Reply:
I just backed-out this change, and re-ran `cmake ..`:

```
-- Found PicoSAT library: /home/Users/avj/solvers/btor_testing_head/boolector/deps/install/lib/libpicosat.so
```

Reply:
(cmake version 3.9.4)

Reply:
@mpreiner I should have read more closely ... that code is guarded by `elseif(NOT PYTHON)`, which is why it doesn't work when building both statically and with `-DPYTHON=ON`.

Reply:
I think I started on a fix for that on one of my branches. Let me check.